src/Application/Actions/ChangePasswordUsingToken.php

```diff
@@ -47,7 +48,7 @@ protected function action(): Response
 
         foreach ($violations as $violation) {
             if ($violation->getPropertyPath() === 'raw_password') {
-                throw new HttpBadRequestException($this->request, $violation->getMessage());
+                throw new HttpBadRequestException($request, (string)$violation->getMessage());
```

Comment:
Is the `(string)` cast _really_ needed here?

Reply:
Based on the available type information it is. `\Slim\Exception\HttpSpecializedException::__construct` declares that it needs a string or null as its second argument, but `\Symfony\Component\Validator\ConstraintViolationInterface::getMessage` declares that it returns `string|\Stringable`. So in case it returns a Stringable we'd have to convert it to a string.

Reply:
It's true we could let PHP do that for us automatically - but I believe it makes things a lot clearer if we cast explicitly when necessary instead of relying on the PHP automatic type juggling.